model/inline/src/main/scala/aqua/model/inline/TagInliner.scala

```diff
@@ -39,58 +39,65 @@ object TagInliner extends Logging {
    *
    * @param prefix Previous instructions
    */
-  enum TagInlined(prefix: Option[OpModel.Tree]) {
+  enum TagInlined[F](prefix: Option[OpModel.Tree]) {
 
     /**
      * Tag inlining emitted nothing
      */
-    case Empty(
+    case Empty[S](
```

Comment:
I believe it is possible to not make cases of enum have their own type parameters. They can use type parameter defined for the whole enum

Reply:
Scala fails to handle default parameters in this case =(